### ISSUE TITLE
torch.special.log_softmax loses a representable float64 VJP component

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#195680]
### 🐛 Describe the bug

`torch.special.log_softmax` returns `0.0` for a first-order derivative component whose mathematically correct value is finite, normal, and nonzero in `float64`.

LogSoftmax is equivariant to a common shift of all logits, so the resulting input VJP should sum to zero. In the example below, the high-precision reference VJP rounds to approximately `[5.551115123125776e-17, -5.551115123125776e-17]`. However, PyTorch collapses the first component to `0.0` while preserving the second component, so the VJP sum is no longer zero.

This issue reproduces on CPU, so CUDA availability does not affect the result.

```python
import torch

dtype = torch.float64

z = torch.tensor([37.42994775023705, 0.0], dtype=dtype, requires_grad=True)
cotangent = torch.tensor([1.0, 0.0], dtype=dtype)

y = torch.special.log_softmax(z, dim=-1)
actual, = torch.autograd.grad(y, z, grad_outputs=cotangent)

expected = torch.tensor(
    [5.551115123125776e-17, -5.551115123125776e-17],
    dtype=dtype,
)

print("actual:")
print(actual)
print("expected:")
print(expected)
print("actual VJP sum:")
print(actual.sum())
print("expected VJP sum:")
print(expected.sum())
```

### Actual vs expected result

```text
actual:
tensor([ 0.0000e+00, -5.5511e-17], dtype=torch.float64)

expected:
tensor([ 5.5511e-17, -5.5511e-17], dtype=torch.float64)

actual VJP sum:
tensor(-5.5511e-17, dtype=torch.float64)

expected VJP sum:
tensor(0., dtype=torch.float64)
```

The discrepancy occurs in the first-order backward pass. The expected first VJP component is a finite, normal, nonzero `float64` value, but PyTorch collapses it to `0.0`.

### Versions

```text
PyTorch version: 2.12.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (x86_64)
Python version: 3.13.13 | packaged by Anaconda, Inc.
Python platform: Linux x86_64

Is CUDA available: False
GPU models:
  GPU 0: NVIDIA RTX 6000 Ada Generation
  GPU 1: NVIDIA RTX 6000 Ada Generation
Nvidia driver version: 570.211.01

CPU:
  Model name: AMD Ryzen Threadripper PRO 7985WX 64-Cores
  CPU(s): 128

Versions of relevant libraries:
[pip3] numpy==2.4.6
[pip3] torch==2.12.0
[pip3] triton==3.7.0
```

cc @ezyang @albanD @gqchen @nikitaved @soulitzer @Varal7 @bobrenjc93
